pkg/metrics/dataset_metrics.go

```diff
@@ -74,4 +83,5 @@ func (m *datasetMetrics) Forget() {
 func init() {
 	metrics.Registry.MustRegister(datasetUFSFileNum, datasetUFSTotalSize)
 	datasetMetricsMap = map[string]*datasetMetrics{}
+	datasetMetricsMutex = &sync.Mutex{}
```

Comment:
Why not using `&sync.RWMutex{}` to separate read and write?

Reply:
The two related operations are insertion and deletion. There is no read operation on metrics map.

Reply:
I see what you mean. RWMutex lgtm, will change it.